### PR TITLE
test(ci): harden web and store e2e checks

### LINKED ISSUE #176: Wire KMP and MAUI FireOS/Horizon e2e build targets
## Context

The expanded `e2e-tests` workflow now requires every library/platform row to be reported explicitly. During that audit, the KMP and MAUI rows for FireOS/Amazon and Horizon had to be marked as blocked because there is no build switch or flavor wiring for those targets yet.

This is different from `react-native-iap`, `expo-iap`, and `flutter_inapp_purchase`, which already expose Android store selection through flags such as `fireOsEnabled`, `horizonEnabled`, `EXPO_IAP_FIREOS`, and `EXPO_IAP_HORIZON`.

## Work to do

- Add FireOS/Amazon and Horizon build wiring for `libraries/kmp-iap`.
- Add FireOS/Amazon and Horizon binding/example wiring for `libraries/maui-iap`.
- Make the target selection explicit so the e2e workflow can build these rows instead of reporting them as blocked.
- Update `.claude/commands/e2e-tests.md` once real commands exist.
- Verify the normal Android/Play and iOS paths still build after adding the new target switches.

## Current blocked rows

```text
KMP FireOS   | BLOCKED | no FireOS/Amazon flavor switch wired
KMP Horizon  | BLOCKED | no Horizon flavor switch wired
MAUI FireOS  | BLOCKED | no Amazon binding/example flavor switch wired
MAUI Horizon | BLOCKED | no Horizon binding/example flavor switch wired
```

## Acceptance criteria

- KMP has documented commands for Android/Play, FireOS/Amazon, Horizon, and iOS e2e build rows.
- MAUI has documented commands for Android/Play, FireOS/Amazon, Horizon, and iOS e2e build rows.
- `e2e-tests` can report these rows as `PASS` or `FAIL` based on real commands, not `BLOCKED: not wired`.

## Follow-up note

Pick this up after PR #162 is merged. There is already in-progress work for these target rows, so the next pass should resume from that work once the FireOS/Vega runtime support from #162 has landed on `main`.

### PR DESCRIPTION
**Pull request by hyochan**
## Summary

- Retry timing-only web performance flakes while preserving byte-budget failures.
- Replace stale KMP/MAUI FireOS/Horizon e2e blocked rows with real store-specific build commands.
- Expand KMP/MAUI CI coverage for store flavor builds and Google package changes.
- Clarify commit workflow branch naming to use semantic prefixes instead of agent/tool prefixes.

## Test plan

- [x] `node --check scripts/e2e-web-sites.mjs`
- [x] `bunx prettier --check scripts/e2e-web-sites.mjs .claude/commands/e2e-tests.md .github/workflows/ci-kmp-iap.yml .github/workflows/ci-maui-iap.yml`
- [x] `actionlint .github/workflows/ci-kmp-iap.yml .github/workflows/ci-maui-iap.yml`
- [x] `./gradlew :library:compilePlayDebugKotlinAndroid :library:compileHorizonDebugKotlinAndroid :library:compileAmazonDebugKotlinAndroid :example:composeApp:assemblePlayDebug :example:composeApp:assembleHorizonDebug :example:composeApp:assembleAmazonDebug` from `libraries/kmp-iap`
- [x] CI-equivalent docs/kit build + preview/server + `bun run e2e:web`
- [x] SDK parity audit via pre-commit

## Preview

No recording attached: this changes CI/e2e workflow behavior and agent test instructions, with no user-facing visual surface. Terminal and CI-equivalent proof is listed above.

## Summary by CodeRabbit

* **New Features**
  * Expanded end-to-end regression guidance to cover FireOS/Amazon and Horizon builds for both KMP and MAUI.
  * Added more store-specific build and install steps for Android checks.
  * Introduced retry handling for performance checks, with clearer success reporting when extra attempts are needed.

* **Bug Fixes**
  * Updated regression status so previously blocked FireOS/Horizon paths now show the correct build-only or build + device-flow expectations.
  * Improved CI coverage to react to a broader set of relevant changes and build more app variants.

Closes #176

### CODE REVIEW

File: scripts/e2e-web-sites.mjs [hunk 1]
```diff
@@ -417,15 +454,71 @@ async function checkPerformance(page, siteName, route) {
...
+      if (
+        !isRetryablePerformanceError(error) ||
+        attempt === PERFORMANCE_ATTEMPTS
+      ) {
+        throw error;
+      }
+    }
+  }
+
+  throw lastError;
```
[the remark this answers is not in the document]
Comment (hyochan) [reply scripts/e2e-web-sites.mjs]: Fixed in cf7537da. Removed the unreachable fallback throw and the now-unused lastError tracking from checkPerformanceWithRetry.

**Comment by hyochan** [comment]
Addressed current automated review feedback in cf7537da and f9a61778.

File: .claude/commands/e2e-tests.md [hunk 2]
```diff
@@ -525,37 +549,108 @@ xcodebuild \
...
 ```bash
 cd packages/google
-./gradlew :openiap:assemblePlayRelease
-cd ../../libraries/maui-iap/android
-../../../packages/google/gradlew :openiap:assembleRelease
+./gradlew \
+  :openiap:assemblePlayRelease \
+  :openiap:assembleAmazonRelease \
+  :openiap:assembleHorizonRelease
 cd ../../..
```
[the remark this answers is not in the document]
Comment (hyochan) [reply .claude/commands/e2e-tests.md]: Fixed in 9be979d7. Corrected the MAUI e2e prerequisite path from packages/google back to the repository root with cd ../.. .

File: scripts/e2e-web-sites.mjs [hunk 3]
```diff
@@ -417,15 +454,66 @@ async function checkPerformance(page, siteName, route) {
...
+  } finally {
+    await session.detach();
+  }
+}
+
+async function checkPerformanceWithRetry(page, siteName, route) {
+  for (let attempt = 1; attempt <= PERFORMANCE_ATTEMPTS; attempt += 1) {
+    if (attempt > 1) {
+      await clearBrowserCache(page);
+      await page.reload({
```
[the remark this answers is not in the document]
Comment (hyochan) [reply scripts/e2e-web-sites.mjs]: Fixed in 0cb6f1fd. Passed the pageErrors accumulator into the performance retry helper and clear it before retry reloads so stale first-attempt runtime errors do not fail a successful retry.

File: scripts/e2e-web-sites.mjs [hunk 4]
```diff
@@ -536,10 +624,16 @@ async function checkSite(browser, request, site) {
         }
 
         if (viewport.name === "desktop") {
-          const metrics = await checkPerformance(page, site.name, route);
+          const { metrics, attempts } = await checkPerformanceWithRetry(
+            page,
+            site.name,
+            route,
+          );
```
[the remark this answers is not in the document]
Comment (hyochan) [reply scripts/e2e-web-sites.mjs:634]: Fixed in 0cb6f1fd. checkSite now passes pageErrors into checkPerformanceWithRetry so the retry path can reset stale errors before reloading.

File: scripts/e2e-web-sites.mjs [hunk 5]
```diff
@@ -417,15 +454,67 @@ async function checkPerformance(page, siteName, route) {
...
+      const metrics = await checkPerformance(page, siteName, route);
+      return { metrics, attempts: attempt };
+    } catch (error) {
+      if (
+        !isRetryablePerformanceError(error) ||
+        attempt === PERFORMANCE_ATTEMPTS
+      ) {
+        throw error;
+      }
+    }
```
[the remark this answers is not in the document]
Comment (hyochan) [reply scripts/e2e-web-sites.mjs:515]: Fixed in 5fc2db4a. The performance retry path now refuses to retry when pageErrors already contains runtime or request failures, so those failures are not cleared and hidden by a later reload.
